control/cli.py

```diff
@@ -296,53 +296,84 @@ def remove_host(self, args):
         argument("-s", "--trsvcid", help="Port number", default="4420", required=False),
     ])
     def create_listener(self, args):
         """Creates a listener for a subsystem at a given IP/Port."""
         traddr = GatewayConfig.escape_address_if_ipv6(args.traddr)
-        req = pb2.create_listener_req(
-            nqn=args.subnqn,
-            gateway_name=args.gateway_name,
-            trtype=args.trtype,
-            adrfam=args.adrfam,
-            traddr=traddr,
-            trsvcid=args.trsvcid,
-        )
-        ret = self.stub.create_listener(req)
-        self.logger.info(f"Created {args.subnqn} listener at {traddr}:{args.trsvcid}: {ret.status}")
+        trtype = None
+        adrfam = None
+        if args.trtype:
+            trtype = args.trtype.upper()
+        if args.adrfam:
+            adrfam = args.adrfam.lower()
+        try:
+            req = pb2.create_listener_req(
+                nqn=args.subnqn,
+                gateway_name=args.gateway_name,
+                trtype=trtype,
+                adrfam=adrfam,
+                traddr=traddr,
+                trsvcid=args.trsvcid,
+            )
+            ret = self.stub.create_listener(req)
+            self.logger.info(f"Created {args.subnqn} listener at {traddr}:{args.trsvcid}: {ret.status}")
+        except ValueError as err:
+            self.logger.error(f"{err}")
+            self.logger.info(f"Created {args.subnqn} listener at {traddr}:{args.trsvcid}: {False}")
+            raise
+        except Exception as ex:
+            self.logger.info(f"Created {args.subnqn} listener at {traddr}:{args.trsvcid}: {False}")
+            raise
 
     @cli.cmd([
         argument("-n", "--subnqn", help="Subsystem NQN", required=True),
         argument("-g", "--gateway-name", help="Gateway name", required=True),
         argument("-t", "--trtype", help="Transport type", default="TCP"),
-        argument("-f", "--adrfam", help="Address family", default="ipv4"),
+        argument("-f", "--adrfam", help="Address family", default="IPV4"),
         argument("-a", "--traddr", help="NVMe host IP", required=True),
         argument("-s", "--trsvcid", help="Port number", default="4420", required=False),
     ])
     def delete_listener(self, args):
         """Deletes a listener from a subsystem at a given IP/Port."""
         traddr = GatewayConfig.escape_address_if_ipv6(args.traddr)
-        req = pb2.delete_listener_req(
-            nqn=args.subnqn,
-            gateway_name=args.gateway_name,
-            trtype=args.trtype,
-            adrfam=args.adrfam,
-            traddr=traddr,
-            trsvcid=args.trsvcid,
-        )
-        ret = self.stub.delete_listener(req)
-        self.logger.info(f"Deleted {traddr}:{args.trsvcid} from {args.subnqn}: {ret.status}")
+        trtype = None
+        adrfam = None
+        if args.trtype:
+            trtype = args.trtype.upper()
+        if args.adrfam:
+            adrfam = args.adrfam.lower()
```

Comment:
in all if validation parts I think it is a good practice to add the else option as well in order the get a better logs in a failure cases:
e.g.
            else:
                self.logger("No .... was provided")

Reply:
This is not needed as these parameters are not required. There is no need to display a message if they're not there.